libs/llrb/llrb.go

```diff
@@ -143,26 +143,26 @@ func (t *llrb) deleteMin(h *node) (*node, *node) {
 func (t *llrb) Remove(key NodeKey) (interface{}, error) {
 	t.mtx.Lock()
 	defer t.mtx.Unlock()
-	var deleted *node
+	deleted := node{}
```

Comment:
no need for this declaration

Reply:
Can we do the assignment to an **existed** and _new_ variable without declaring a new variable?
**t.root**, _deleted_ = t.delete(t.root, key)